crates/fuel-core/src/service/adapters/compression_adapters.rs

```diff
@@ -21,8 +23,8 @@ impl block_source::BlockSource for BlockImporterAdapter {
         use futures::StreamExt;
         self.events_shared_result()
             .map(|result| {
-                let sealed_block = result.sealed_block.clone();
-                let events = result.events.clone();
+                let sealed_block = result.deref().as_ref().sealed_block.clone();
```

Comment:
```suggestion
                let sealed_block = (*result).as_ref().sealed_block.clone();
```

Explicit `.deref()` makes my eyes hurt :)

Reply:
Interesting. I prefer the explicit over the *-operator. But I'll give you that the operator is more concise. I'll use it more in the future.